app/src/screens/ProposalDetail.js

```diff
@@ -48,43 +59,104 @@ function ProposalDetail({ proposal, onBack, requestToken }) {
     threshold,
   } = proposal
 
-  const myStakes = stakes.filter(({ entity }) =>
-    addressesEqual(entity, connectedAccount)
+  const myStake = useMemo(
+    () => stakes.find(({ entity }) => addressesEqual(entity, connectedAccount)) || {amount: new BigNumber('0')},
+    [stakes]
   )
-  const didIStaked = myStakes.length > 0 && [...myStakes].pop().tokensStaked > 0
+
+
+  const myStakeAmountFormatted = useMemo(() => {
+    if (!myStake) {
+      return '0'
+    }
```

Comment:
We could remove this since we are default to  `{ amount: '0' }` right? 
Consider also removing the `useMemo`